src/main/java/gregtech/api/capability/impl/AbstractRecipeLogic.java

```diff
@@ -140,25 +153,26 @@ protected void updateRecipeProgress() {
 
     protected void trySearchNewRecipe() {
         long maxVoltage = getMaxVoltage();
-        Recipe currentRecipe = null;
         IItemHandlerModifiable importInventory = getInputInventory();
         IMultipleTankHandler importFluids = getInputTank();
-        if (previousRecipe != null && previousRecipe.matches(false, importInventory, importFluids)) {
+        if (this.previousRecipe != null && this.previousRecipe.matches(false, importInventory, importFluids)) {
             //if previous recipe still matches inputs, try to use it
-            currentRecipe = previousRecipe;
+            this.currentRecipe = this.previousRecipe;
         } else {
             boolean dirty = checkRecipeInputsDirty(importInventory, importFluids);
-            if (dirty || forceRecipeRecheck) {
+            if (dirty || this.forceRecipeRecheck) {
                 this.forceRecipeRecheck = false;
                 //else, try searching new recipe for given inputs
-                currentRecipe = findRecipe(maxVoltage, importInventory, importFluids);
-                if (currentRecipe != null) {
-                    this.previousRecipe = currentRecipe;
+                this.currentRecipe = findRecipe(maxVoltage, importInventory, importFluids);
+                if (this.currentRecipe != null) {
+                    this.previousRecipe = this.currentRecipe;
                 }
             }
         }
-        if (currentRecipe != null && setupAndConsumeRecipeInputs(currentRecipe)) {
-            setupRecipe(currentRecipe);
+        if (this.currentRecipe != null && setupAndConsumeRecipeInputs(this.currentRecipe)) {
+            setupRecipe(this.currentRecipe);
+        } else {
+            metaTileEntity.setSituation(OUTPUT_INVENTORY_FULL);
```

Comment:
As a note, the `trySearchNewRecipe` method is overridden in `MetaTileEntityMultiFurnace`, so it would not benefit from this output full situation being set here. The situation being set should also be added to `MetaTileEntityMultiFurnace`.